simple-stack/src/main/java/com/zhuinden/simplestack/ScopeManager.java

```diff
@@ -330,9 +331,17 @@ void setGlobalServices(GlobalServices globalServices) {
         this.globalServices = globalServices;
     }
 
+    void setGlobalServices(GlobalServices.Factory globalServiceFactory) {
+        this.globalServiceFactory = globalServiceFactory;
+    }
+
 
     private void buildGlobalScope() {
         if(!scopes.containsKey(GLOBAL_SCOPE_TAG)) {
+            if(globalServiceFactory != null) {
+                globalServices = globalServiceFactory.create();
```

Comment:
@matejdro do you think this variant makes sense, or am I supposed to try to merge a factory with the non-factory variant?

Maybe explode in `.install()` saying "cannot set both a `GlobalServices` and a `GlobalServices.Factory` at the same time? 🤔

Reply:
Oh, wait, you've already answered this:

>  You can leave old setGlobalServices method in. This method would **just create dummy Factory that would return the provided value.**

Nevermind, I'll look into that approach instead 🤔

Reply:
I can't help but think that's functionally equivalent to this one.

Reply:
It is functionally equivalent, but the dummy factory one is a bit cleaner IMO since you need one less field and less if checks in Navigator.Installer.